Tests/SonarScanner.MSBuild.PreProcessor.Test/ArgumentProcessorTests.cs

```diff
@@ -495,6 +495,34 @@ public void PreArgProc_Timeout()
             CheckProcessingSucceeds("/key:k1", "/d:sonar.http.timeout=0").HttpTimeout.Should().Be(TimeoutProvider.DefaultHttpTimeout);
         }
 
+        [DataTestMethod]
+        [DataRow("C:\\Program Files\\Java\\jdk1.6.0_30\\bin\\javac.exe", "C:\\Program Files\\Java\\jdk1.6.0_30\\bin\\javac.exe")]
+        [DataRow("C:Program Files\\Java\\jdk1.6.0_30\\bin\\javac.exe", "C:Program Files\\Java\\jdk1.6.0_30\\bin\\javac.exe")]
+        [DataRow("\\jdk1.6.0_30\\bin\\javac.exe", "\\jdk1.6.0_30\\bin\\javac.exe")]
+        [DataRow("jdk1.6.0_30\\bin\\javac.exe", null)]
+        [DataRow("not a path", null)]
+        [DataRow(" ", null)]
+        public void PreArgProc_JavaExePath_Set(string javaExePath, string result) =>
+            CheckProcessingSucceeds("/k:key", $"/d:sonar.scanner.javaExePath={javaExePath}").JavaExePath.Should().Be(result);
```

Comment:
Processing needs to fail with an error if we ignore the user setting (`null` cases).

Reply:
So the parameter becomes mandatory?

Reply:
No. If it is set to something we do not respect, we need to fail instead of ignoring the value provided.